x-pack/platform/plugins/private/upgrade_assistant/server/lib/es_deprecations_status/index.ts

```diff
@@ -6,20 +6,71 @@
  */
 
 import { IScopedClusterClient } from '@kbn/core/server';
-import { EnrichedDeprecationInfo, ESUpgradeStatus, FeatureSet } from '../../../common/types';
+import {
+  EnrichedDeprecationInfo,
+  ESUpgradeStatus,
+  FeatureSet,
+  DataSounceExclusions,
+  DataStreamsAction,
+  ReindexAction,
+} from '../../../common/types';
 import { getEnrichedDeprecations } from './migrations';
 import { getHealthIndicators } from './health_indicators';
+import { matchExclusionPattern } from '../data_source_exclusions';
 
 export async function getESUpgradeStatus(
   dataClient: IScopedClusterClient,
-  featureSet: FeatureSet
+  {
+    featureSet,
+    dataSourceExclusions,
+  }: { featureSet: FeatureSet; dataSourceExclusions: DataSounceExclusions }
 ): Promise<ESUpgradeStatus> {
   const getCombinedDeprecations = async () => {
     const healthIndicators = await getHealthIndicators(dataClient);
     const enrichedDeprecations = await getEnrichedDeprecations(dataClient);
 
-    const toggledMigrationsDeprecations = enrichedDeprecations.filter(
-      ({ type, correctiveAction }) => {
+    const toggledMigrationsDeprecations = enrichedDeprecations
+      .map((deprecation) => {
+        const correctiveActionType = deprecation.correctiveAction?.type;
+        if (correctiveActionType === 'dataStream') {
+          const excludedActions = matchExclusionPattern(deprecation.index!, dataSourceExclusions);
+
+          (deprecation.correctiveAction as DataStreamsAction).metadata.excludedActions =
+            excludedActions;
+        }
+        if (correctiveActionType === 'reindex') {
```

Comment:
```suggestion
        } else if (correctiveActionType === 'reindex') {
```
extreme nit: but I'd change these logic blocks together